ci/azure-pipelines.yml

```diff
@@ -129,47 +122,64 @@ stages:
       - job: snapshot_publish
         dependsOn: main
         condition: and(succeeded(),eq(variables['Build.Reason'], 'IndividualCI'))
+        variables:
+          PACKAGE_VERSION: $[ dependencies.main.outputs['BuildData.PACKAGE_VERSION'] ]
         steps:
           - task: DownloadPipelineArtifact@2
             inputs:
-              artifact: java-docker-image
+              artifact: javaenv-docker-image
               path: $(Build.SourcesDirectory)/build
           - script: |
-              docker image load --input build/fabric-javaenv.tar.gz         # gets the build image of nodeenv
+              docker image load --input build/fabric-javaenv.tar.gz
               docker images
               # Publish docker images to nexus repository
               docker login ${DOCKER_REGISTRY_URL} --username=${DOCKER_REGISTRY_USERNAME} --password=${DOCKER_REGISTRY_PASSWORD}
               echo "Logged in to docker registry"
               # tag javaenv image to $PUSH_VERSION
-              docker tag hyperledger/fabric-javaenv ${DOCKER_REGISTRY_URL}/fabric-javaenv:amd64-${MAPPED_VERSION}
+              docker tag hyperledger/fabric-javaenv ${DOCKER_REGISTRY_URL}/fabric-javaenv:amd64-${MAPPED_VERSION}-stable
               # push javaenv to nexus repository
-              docker push ${DOCKER_REGISTRY_URL}/fabric-javaenv:amd64-${MAPPED_VERSION}
-            env: 
-              DOCKER_REGISTRY_USERNAME: $(GITHUB_REGISTRY_USER)
-              DOCKER_REGISTRY_PASSWORD: $(GITHUB_REGISTRY_PWD_TOKEN)
-              DOCKER_REGISTRY_URL: docker.pkg.github.com/hyperledger/fabric-chaincode-java
-              MAPPED_VERSION:  $(PUSH_VERSION)
+              docker push ${DOCKER_REGISTRY_URL}/fabric-javaenv:amd64-${MAPPED_VERSION}-stable
+            env:
+              DOCKER_REGISTRY_USERNAME: $(nexus-user)
+              DOCKER_REGISTRY_PASSWORD: $(nexus-password)
+              DOCKER_REGISTRY_URL: nexus3.hyperledger.org:10003/hyperledger
+              MAPPED_VERSION: $(PACKAGE_VERSION)
+            displayName: 'docker push to nexus'
 
   # As the next script is more complex and uses loops, run this descretely in a sh file
   # Publishing step for git tags
   - stage: Publish_tag
     condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))
     jobs:
-      - job: update_version
-        steps:
-          - script: |
-              env | sort
-              echo "Update the version number"
-              echo "Make sure release notes are present"
-              echo "Make sure change history is present"
-      - job: npm_publish
+      - job: publish_release
         steps:
+          - template: templates/build-data.yml
```

Comment:
Is this line needed here?

Reply:
oh it's used lower down... is the information from the previous stage not available here?

Reply:
You can't share variables between stages without getting in to REST calls to the azure pipeline API as far as I can tell. A template seemed way simpler!

Reply:
ah well in that case - this seems like the best solution.